node/src/components/block_proposer.rs

```diff
@@ -261,38 +249,33 @@ impl BlockProposerReady {
                             request.random_bit,
                         ))
                         .ignore()
+                } else {
+                    Effects::new()
```

Comment:
We should log a warning here. It probably means that a block request spent so much time in the reactor queue (or in the block proposer queue?) that in the meantime we finalized a block with a timestamp newer than this.